contracts/infinity-pool/src/testing/tests/sim_tests/sim_direct_swap_nfts_for_tokens.rs

```diff
@@ -0,0 +1,733 @@
+        nft_token_ids: vec![token_id_1.to_string(), token_id_2.to_string()],
+    };
+    let _ = router.execute_contract(user1.clone(), infinity_pool.clone(), &msg, &[]);
+
+    // Owner can deposit tokens
+    let deposit_amount_1 = deposit_amount;
+    let msg = ExecuteMsg::DepositTokens { pool_id: pool.id };
+    let res = router.execute_contract(
+        creator,
+        infinity_pool.clone(),
+        &msg,
+        &coins(deposit_amount_1, NATIVE_DENOM),
+    );
+    match res {
+        Ok(_) => Ok(DepositNftsResult {
+            infinity_pool,
+            token_id_1,
+            token_id_2,
+        }),
+        Err(err) => Err(err),
+    }
+}
+
+fn get_sim_swap_message(
+    pool: Pool,
+    token_id_1: u32,
+    token_amount: u128,
+    robust: bool,
+    user2: Addr,
+    finder: Option<String>,
+) -> msg::QueryMsg {
+    SimDirectSwapNftsForTokens {
+        pool_id: pool.id,
+        nfts_to_swap: vec![NftSwap {
+            nft_token_id: token_id_1.to_string(),
+            token_amount: Uint128::new(token_amount),
+        }],
+        swap_params: SwapParams {
+            deadline: Timestamp::from_nanos(GENESIS_MINT_START_TIME).plus_seconds(1000),
+            robust,
+        },
+        token_recipient: user2.to_string(),
+        finder,
+    }
+}
+
+fn set_pool_active(
+    router: &mut StargazeApp,
+    is_active: bool,
+    pool: Pool,
+    creator: Addr,
+    infinity_pool: Addr,
+) {
+    let msg = ExecuteMsg::SetActivePool {
+        is_active,
+        pool_id: pool.id,
+    };
+    let _ = router.execute_contract(creator, infinity_pool, &msg, &[]);
+}
+
+fn check_nft_sale(
+    expected_spot_price: u128,
+    expected_royalty_price: u128,
+    expected_network_fee: u128,
+    expected_finders_fee: u128,
+    swaps: Vec<Swap>,
+    pool: Pool,
+    creator: Addr,
+    user2: Addr,
+    token_id: String,
+) {
+    assert_eq!(swaps[0].pool_id, pool.id);
+    assert_eq!(swaps[0].spot_price.u128(), expected_spot_price);
+    println!("swaps is {:?}", swaps);
+
+    let expected_nft_payment = Some(NftPayment {
+        nft_token_id: token_id,
+        address: ASSET_ACCOUNT.to_string(),
+    });
+    assert_eq!(swaps[0].nft_payment, expected_nft_payment);
+    let expected_royalty_payment = Some(TokenPayment {
+        amount: Uint128::new(expected_royalty_price),
+        address: creator.to_string(),
+    });
+    assert_eq!(swaps[0].royalty_payment, expected_royalty_payment);
+    let network_fee = swaps[0].network_fee.u128();
+    println!(
+        "network fee: {:?} expected_network_fee: {}",
+        network_fee, expected_network_fee
+    );
+    assert_eq!(network_fee, expected_network_fee);
+    let mut expected_price = expected_spot_price - network_fee;
+    expected_price -= expected_royalty_payment.unwrap().amount.u128();
+    println!(
+        "expected price {:?} expected finders fee {:?}",
+        expected_price, expected_finders_fee
+    );
+    expected_price -= expected_finders_fee;
+
```

Comment:
I am constructing this because this is part of the return from the swaps command. I think finders_fee just has a price instead of it being a full fledged TokenPayment object, but i'll check again.

Reply:
I am now checking all three in the other PR (https://github.com/public-awesome/infinity/pull/3): 
```rust

    let expected_seller_payment = Some(TokenPayment {
        amount: Uint128::new(expected_price),
        address: user2.to_string(),
    });
    assert_eq!(swaps[0].seller_payment, expected_seller_payment);

    let expected_finder_payment = match expected_finders_fee {
        0 => None,
        _ => Some(TokenPayment {
            amount: Uint128::from(expected_finders_fee),
            address: user2.to_string(),
        }),
    };
    assert_eq!(swaps[0].finder_payment, expected_finder_payment);

    let expected_royalty_payment = Some(TokenPayment {
        amount: Uint128::new(expected_royalty_price),
        address: creator.to_string(),
    });
    assert_eq!(swaps[0].royalty_payment, expected_royalty_payment);
    
   ```